src/parser/core/Events.ts

```diff
@@ -380,7 +380,7 @@ export interface ApplyBuffStackEvent extends BuffEvent<EventType.ApplyBuffStack>
 }
 
 export interface ApplyDebuffStackEvent extends BuffEvent<EventType.ApplyDebuffStack> {
-  sourceID: number;
+  targetInstance?: number;
```

Comment:
The git diff looks a little funky, this file changed while I was working on it.  BuffEvent has sourceID already, nothing is lost here.

Reply:
They are not quite the same -- in BuffEvent sourceID is optional, and ApplyDebuffStackEvent overrides that to say that it is required. The question comes down to whether or not we do know ApplyDebuffStack always contains a sourceID or not

Reply:
Hmm, what are examples of BuffEvents that don't have a sourceID?

Reply:
I guess if we weren't getting errors before won't hurt to readd sourceID as required

Reply:
instead of an ID it sometimes gives you a source object; for example Wind Rush Totem applybuff for the speed buff will ```
show up as 
```typescript
{ 
 timestamp: 1961973;
 ability: { [ability info..] };
 source: {name: 'Wind Rush Totem', ...} 
}
```

With no sourceID :(